src/storage/txn/scheduler.rs

```diff
@@ -199,7 +199,7 @@ struct SchedulerInner<L: LockManager> {
     // used to control write flow
     running_write_bytes: CachePadded<AtomicUsize>,
 
-    flow_controller: Arc<FlowController>,
+    flow_controller: Arc<dyn FlowController + Send + Sync>,
```

Comment:
+1

Reply:
Generic will introduce too much code change (As it will add additional generic parameter for all places that use this class). Just like TabletFactory, I prefer dynamic dispatch here. The performance difference is very very little between two.

Reply:
we already have a generic LockManager, the change won't be too breaking

Reply:
There's other places using Arc<FlowController>,  such as  node.rs, config.rs, config_manager.rs, mod.rs.  The performance difference of dyn and static binding is negligible in this case, the actual dispatch impact is at most a few ns per call.

Reply:
Perhaps we can use `enum` in this case. Flow control should work on per tablet level, so if there is only one tablet, then it can use singleton, otherwise, it's upgraded to multiple tablets. This is also how `mpsc` implemented in std lib and other channel libraries.